Content.Server/GameTicking/GamePresets/PresetTraitor.cs

```diff
@@ -0,0 +1,210 @@
+                }
+
+                mind.AddRole(traitorRole);
+                _traitors.Add(traitorRole);
+                pdaComponent.InitUplinkAccount(uplinkAccount);
+            }
+
+            var codewordPool = new List<string>(Codewords);
+            var finalCodewordCount = Math.Min(CodewordCount, Codewords.Length);
+            string[] codewords = new string[finalCodewordCount];
+            for (int i = 0; i < finalCodewordCount; i++)
+            {
+                codewords[i] = _random.PickAndTake(codewordPool);
+            }
+
+            foreach (var traitor in _traitors)
+            {
+                traitor.GreetTraitor(codewords);
+            }
+
+            _gameticker.AddGameRule<RuleTraitor>();
+            return true;
+        }
+
+        public override void OnGameStarted()
+        {
+            var objectivesMgr = IoCManager.Resolve<IObjectivesManager>();
+            foreach (var traitor in _traitors)
+            {
+                //give traitors their objectives
+                var difficulty = 0f;
+                for (var pick = 0; pick < MaxPicks && MaxDifficulty > difficulty; pick++)
+                {
+                    var objective = objectivesMgr.GetRandomObjective(traitor.Mind);
+                    if (objective == null) continue;
+                    if (traitor.Mind.TryAddObjective(objective))
+                        difficulty += objective.Difficulty;
+                }
+            }
+        }
+
+        public override string GetRoundEndDescription()
+        {
+            var traitorCount = _traitors.Count;
+            var result = Loc.GetString("There {0} {1} {2}.", Loc.GetPluralString("was", "were", traitorCount),
+                traitorCount, Loc.GetPluralString("traitor", "traitors", traitorCount));
+            foreach (var traitor in _traitors)
+            {
+                result += Loc.GetString("\n{0} was a traitor",traitor.Mind.Session.Name);
+                var objectives = traitor.Mind.AllObjectives.ToArray();
+                if (objectives.Length == 0)
+                {
+                    result += ".\n";
+                    continue;
+                }
+
+                result += Loc.GetString(" and had the following objectives:");
+                foreach (var objectiveGroup in objectives.GroupBy(o => o.Prototype.Issuer))
+                {
+                    result += Loc.GetString("\n[color=#87cefa]{0}[/color]", objectiveGroup.Key);
+                    foreach (var objective in objectiveGroup)
+                    {
+                        foreach (var condition in objective.Conditions)
+                        {
+                            var progress = condition.Progress;
+                            result +=
+                                Loc.GetString("\n- {0} | {1}", condition.Title, (progress > 0.99f ? Loc.GetString("[color=green]Success![/color]") : Loc.GetString("[color=red]Failed![/color] ({0}%)", (int) (progress * 100))));
```

Comment:
You could also just translate "Success!" and "Failed!" instead of the stuff around that